rust: Bump `hyper` dependencies

GodTamIt opened the pull request:
## Motivation

Dependents of `svix` in Rust may use newer versions of `hyper` and `rustls` (a dependency in the chain). This helps the package keep up with the latest version so that older versions of these dependent libraries don't have to be compiled.

## Solution

Bumps `hyper` dependencies to the latest versions.

svix-jplatte commented:
I've looked into it and fixed the issue here: https://github.com/svix/svix-webhooks/tree/jplatte/bump-hyper
I think you haven't enabled the "allow maintainers to push to the PR branch" setting on the PR, otherwise I could push directly to the PR.

Could you please update the PR to include that change and make sure the fourth commit is gone (either by adding a revert commit as I did, or by erasing it via interactive rebase)?
Alternatively, I can also open a separate PR and close this one in favor of it (authorship attribution will still be there).

GodTamIt commented:
Okay so I...

1. Rebased the branch on top of `main` ToT
2. Cherry-picked your change for fallback
3. Updated `Cargo.lock` files after the rebase
